modules/tensor_mechanics/test/tests/dynamics/time_integration/ad_newmark_action.i

```diff
@@ -0,0 +1,133 @@
+# Test for  Newmark time integration
+
+# The test is for an 1D bar element of  unit length fixed on one end
+# with a ramped pressure boundary condition applied to the other end.
+# beta and gamma are Newmark time integration parameters
+# The equation of motion in terms of matrices is:
+#
+# M*accel + K*disp = P*Area
+#
+# Here M is the mass matrix, K is the stiffness matrix, P is the applied pressure
+#
+# This equation is equivalent to:
+#
+# density*accel + Div Stress = P
+#
+# The first term on the left is evaluated using the Inertial force kernel
+# The last term on the left is evaluated using StressDivergenceTensors
+# The residual due to Pressure is evaluated using Pressure boundary condition
+
+[Mesh]
+  type = GeneratedMesh
+  dim = 3
+  xmax = 0.1
+  ymax = 1.0
+  zmax = 0.1
+  use_displaced_mesh = false
+[]
+
+[GlobalParams]
+  displacements = 'disp_x disp_y disp_z'
+[]
+
+[Modules/TensorMechanics/DynamicMaster]
+  [all]
+    add_variables = true
+    newmark_beta = 0.25
+    newmark_gamma = 0.5
+    strain = SMALL
+    density = 7750
+    generate_output = 'stress_yy strain_yy'
+    use_automatic_differentiation = true
+  []
+[]
+
+[BCs]
+  [top_x]
+    type = DirichletBC
+    variable = disp_x
+    boundary = top
+    value = 0.0
+  []
+  [top_y]
+    type = DirichletBC
+    variable = disp_y
+    boundary = top
+    value = 0.0
+  []
+  [top_z]
+    type = DirichletBC
+    variable = disp_z
+    boundary = top
+    value = 0.0
+  []
```

Comment:
BCs are not AD here?

Reply:
Not really necessary here (small strain) and I'm testing the AD action here. But if you reeeeeeeeeeeeeeeeeeeeeeeeeeeeealy want I can change it ;-)

Reply:
No, makes sense.